src/Http/Http.Extensions/gen/RequestDelegateGeneratorSources.cs

```diff
@@ -539,8 +535,29 @@ namespace Microsoft.AspNetCore.Http.Generated
     {{GeneratedCodeAttribute}}
     file static class GeneratedRouteBuilderExtensionsCore
     {
-{{GetGenericThunks(genericThunks)}}
-{{GetThunks(thunks)}}
+        private static readonly string[] GetVerb = new[] { Microsoft.AspNetCore.Http.HttpMethods.Get };
+        private static readonly string[] PostVerb = new[] { Microsoft.AspNetCore.Http.HttpMethods.Post };
+        private static readonly string[] PutVerb = new[]  { Microsoft.AspNetCore.Http.HttpMethods.Put };
+        private static readonly string[] DeleteVerb = new[] { Microsoft.AspNetCore.Http.HttpMethods.Delete };
+        private static readonly string[] PatchVerb = new[] { Microsoft.AspNetCore.Http.HttpMethods.Patch };
+
+        {{endpoints}}
+
+        internal static RouteHandlerBuilder MapCore(
+            this IEndpointRouteBuilder routes,
+            string pattern,
+            Delegate handler,
+            IEnumerable<string>? httpMethods,
+            MetadataPopulator populateMetadata,
+            RequestDelegateFactoryFunc createRequestDelegate)
+        {
+            return RouteHandlerServices.Map(routes, pattern, handler, httpMethods, populateMetadata, createRequestDelegate);
+        }
+
+        private static T Cast<T>(Delegate d, T _) where T : Delegate
```

Comment:
Now that we are no longer using strongly-typed delegate types in the overload we need to get a concrete type from the `Delegate` parameter so that we can use it in the generated code. This `Cast` method allows us to capture the concrete type and to capture default parameters passed to the lambda/method group (see [here](https://github.com/dotnet/csharplang/blob/main/proposals/lambda-method-group-defaults.md#conversion-and-unification-behavior)).

Reply:
I think it would be good to add a comment here as well to fully explain this since its kind of obscure (just to help future maintainers).